warrior/Framework/ClassUtils/warrior_connect_class.py

```diff
@@ -409,7 +409,12 @@ def connect_ssh(self):
         # delete -o StrictHostKeyChecking=no and put them in conn_options
         if not conn_options or conn_options is None:
             conn_options = ""
-        command = 'ssh -p {0} {1}@{2} {3}'.format(self.port, self.username,
+        if not self.username:
+            self.username = ""
+            print_info("Username not provided, taking {0} as Username" .format(os.getlogin()))
```

Comment:
Instead of os.getlogin(), kindly use pwd.getpwuid(os.geteuid()).pw_name or pwd.getpwuid(os.geteuid()).[0] as the latter would get the process' real user id.

Reply:
Addressed @brettleehari